src/classes/species.cpp

```diff
@@ -223,3 +224,68 @@ int Species::nCoordinateSets() const { return coordinateSets_.size(); }
 
 // Return coordinates sets
 const std::vector<std::vector<Vec3<double>>> &Species::coordinateSets() const { return coordinateSets_; }
+
+toml::basic_value<toml::discard_comments, std::map, std::vector> Species::serialize()
+{
+    toml::basic_value<toml::discard_comments, std::map, std::vector> species;
+    if (forcefield_ != nullptr)
+        species["forcefield"] = forcefield_->name().data();
+
+    if (atoms_.size() > 0)
+    {
+        toml::array atoms;
+        for (auto &atom : atoms_)
+            atoms.push_back(atom.serialize());
+        species["atom"] = atoms;
+    }
+
+    if (bonds_.size() > 0)
+    {
+        toml::array bonds;
+        for (auto &bond : bonds_)
+            bonds.push_back(bond.serialize());
+        species["bond"] = bonds;
+    }
+
+    if (angles_.size() > 0)
+    {
+        toml::array angles;
+        for (auto &angle : angles_)
+            angles.push_back(angle.serialize());
+        species["angle"] = angles;
+    }
+
+    if (impropers_.size() > 0)
+    {
+        toml::array impropers;
+        for (auto &improper : impropers_)
+            impropers.push_back(improper.serialize());
+        species["improper"] = impropers;
+    }
+
+    if (torsions_.size() > 0)
+    {
+        toml::array torsions;
+        for (auto &torsion : torsions_)
+            torsions.push_back(torsion.serialize());
+        species["torsion"] = torsions;
+    }
+
+    if (isotopologues_.size() > 0)
+    {
+        toml::basic_value<toml::discard_comments, std::map, std::vector> isotopologues;
+        for (auto &isotopologue : isotopologues_)
+            isotopologues[isotopologue->name().data()] = isotopologue->serialize();
+        species["isotopologues"] = isotopologues;
+    }
+
+    if (sites_.size() > 0)
+    {
+        toml::basic_value<toml::discard_comments, std::map, std::vector> sites;
+        for (auto &site : sites_)
+            sites[site.get()->name().data()] = site.get()->serialize();
```

Comment:
```suggestion
            sites[site->name().data()] = site->serialize();
```

Use shared pointer directly